### PR TITLE
Volume Group Datasource

### PR DESCRIPTION
**Pull request by prajwalpatil25**
# Description
Implemented Volume Group data source read operation and added acceptance tests.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [X] Acceptance tests
image

### CODE REVIEW

File: examples/data-sources/powerstore_volumegroup/variables.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+variable "timeout" {
+  type=string
+  description = "Stores the timeout of PowerStore host."
+}
+
+variable "endpoint" {
+  type=string
+  description = "Stores the endpoint of PowerStore host. eg: https://10.1.1.1:443 where 443 is port where API requests are getting accepted"
```
Comment (shenda1) [inline examples/data-sources/powerstore_volumegroup/variables.tf]: do we have to include /api/rest in the endpoint?

File: powerstore/datasource_volume_group.go [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+							MarkdownDescription: "Volume group description.",
+							Computed:            true,
+						},
+						"creation_timestamp": schema.StringAttribute{
+							Description:         "The time at which the volume group was created.",
+							MarkdownDescription: "The time at which the volume group was created.",
+							Computed:            true,
+						},
+						"is_protectable": schema.BoolAttribute{
+							Description:         "This is a derived field that is set internally.",
```
Comment (shenda1) [inline powerstore/datasource_volume_group.go:92]: Description and MarkDownDescription is different ?

File: powerstore/datasource_volume_group_test.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
+package powerstore
+
```
Comment (shenda1) [inline powerstore/datasource_volume_group_test.go:2]: Please add more tests like providing both ID and name, invalid ID

File: powerstore/datasource_volume_group.go [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+							Description:         "Type of volume.",
+							MarkdownDescription: "Type of volume.",
+							Computed:            true,
+						},
+						"is_replication_destination": schema.BoolAttribute{
+							Description:         "Indicates whether this volume group is a replication destination.",
+							MarkdownDescription: "Indicates whether this volume group is a replication destination.",
+							Computed:            true,
+						},
+						"protection_data": schema.ObjectAttribute{
```
Comment (shenda1) [inline powerstore/datasource_volume_group.go:126]: Does MapAttribute fit this parameter?

File: powerstore/datasource_volume_group.go (hunk 2 above)
Comment (prajwalpatil25) [reply powerstore/datasource_volume_group.go:92]: reverted

File: powerstore/datasource_volume_group_test.go (hunk 3 above)
Comment (prajwalpatil25) [reply powerstore/datasource_volume_group_test.go:2]: done

File: examples/data-sources/powerstore_volumegroup/variables.tf (hunk 1 above)
Comment (prajwalpatil25) [reply examples/data-sources/powerstore_volumegroup/variables.tf]: updated description for endpoint

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Review by shenda1 — APPROVED** [review]

**Pull request merged**